### PR TITLE
[Docs] Update new design language page for `v12.0.0-beta.2` release guidance

### PR DESCRIPTION
**Pull request by laurkim**
> [!Important]
> Do not merge until `v12.0.0-beta.2` released

### WHY are these changes introduced?

Updates content on `/new-design-language` page for the `v12.0.0-beta.2` release. Because our migration guidance isn't published on our site yet, I've linked to the guidance for migrations from our `next` branch.

### WHAT is this pull request doing?

Updates documentation specifically for `v12.0.0-beta.2` release.
    <details>
      <summary>What's new page update</summary>
      What's new page update
    </details>
    <details>
      <summary>Timeline update</summary>
      Timeline update
    </details>
    <details>
      <summary>Want to dive in early update</summary>
      Want to dive in early update
    </details>

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by sophschneider — APPROVED** [review]

File: polaris.shopify.com/content/new-design-language/index.md [hunk 1]
```diff
@@ -66,18 +66,34 @@ You will need to review your app's UI and manually patch fixes.
 To prepare for Polaris version 12 make sure you're on the latest version 11.
 It's quick to update from v10 to v11 with our [migrator tool](https://polaris.shopify.com/tools/polaris-migrator) and [migration guide](https://github.com/Shopify/polaris/blob/main/documentation/guides/migrating-from-v10-to-v11.md).
 
-To install the v12 beta version run the following command:
+To install the v12 beta version that **does not require migrations** run the following command:
 
 ```sh
-npm install @shopify/polaris@beta
+npm install @shopify/polaris@beta.1
```
Comment (sophschneider) [inline polaris.shopify.com/content/new-design-language/index.md]: Should this be beta.0? whatever version before we removed the flag

Comment (laurkim) [inline polaris.shopify.com/content/new-design-language/index.md]:
```suggestion
npm install @shopify/polaris@beta.0
```

File: polaris.shopify.com/content/new-design-language/index.md [hunk 2]
```diff
@@ -66,18 +66,34 @@ You will need to review your app's UI and manually patch fixes.
...
 ```sh
-npm install @shopify/polaris@beta
+npm install @shopify/polaris@beta.0
 ```
 
 Or, if you prefer yarn, use the following command instead:
 
 ```sh
-yarn add @shopify/polaris@beta
+yarn add @shopify/polaris@beta.1
```
Comment (laurkim) [inline polaris.shopify.com/content/new-design-language/index.md]:
```suggestion
yarn add @shopify/polaris@beta.0
```

File: polaris.shopify.com/content/new-design-language/index.md [hunk 3]
```diff
@@ -66,18 +66,34 @@ You will need to review your app's UI and manually patch fixes.
 To prepare for Polaris version 12 make sure you're on the latest version 11.
 It's quick to update from v10 to v11 with our [migrator tool](https://polaris.shopify.com/tools/polaris-migrator) and [migration guide](https://github.com/Shopify/polaris/blob/main/documentation/guides/migrating-from-v10-to-v11.md).
 
-To install the v12 beta version run the following command:
+To install the v12 beta version that **does not require migrations** run the following command:
 
 ```sh
-npm install @shopify/polaris@beta
+npm install @shopify/polaris@beta.0
```
Comment (laurkim) [inline polaris.shopify.com/content/new-design-language/index.md]:
```suggestion
npm install @shopify/polaris@12.0.0-beta.0
```

File: polaris.shopify.com/content/new-design-language/index.md [hunk 4]
```diff
@@ -66,18 +66,34 @@ You will need to review your app's UI and manually patch fixes.
...
 ```
 
 Or, if you prefer yarn, use the following command instead:
 
 ```sh
-yarn add @shopify/polaris@beta
+yarn add @shopify/polaris@beta.0
```
Comment (laurkim) [inline polaris.shopify.com/content/new-design-language/index.md:78]:
```suggestion
yarn add @shopify/polaris@12.0.0-beta.0
```

File: polaris.shopify.com/content/new-design-language/index.md [hunk 5]
```diff
@@ -66,18 +66,34 @@ You will need to review your app's UI and manually patch fixes.
...
 ```
 
+To install the v12 beta version that **requires migrations** run the following command:
+
+```sh
+npm install @shopify/polaris@beta.2
```
Comment (laurkim) [inline polaris.shopify.com/content/new-design-language/index.md]:
```suggestion
npm install @shopify/polaris@12.0.0-beta.2
```

File: polaris.shopify.com/content/new-design-language/index.md [hunk 6]
```diff
@@ -66,18 +66,34 @@ You will need to review your app's UI and manually patch fixes.
...
+```
+
+Or, if you prefer yarn, use the following command instead:
+
+```sh
+yarn add @shopify/polaris@beta.2
```
Comment (laurkim) [inline polaris.shopify.com/content/new-design-language/index.md]:
```suggestion
yarn add @shopify/polaris@12.0.0-beta.2
```

File: polaris.shopify.com/content/new-design-language/index.md [hunk 7]
```diff
@@ -66,18 +66,34 @@ You will need to review your app's UI and manually patch fixes.
...
 ```sh
-npm install @shopify/polaris@beta
+npm install @shopify/polaris@12.0.0-beta.0
 ```
 
 Or, if you prefer yarn, use the following command instead:
 
 ```sh
-yarn add @shopify/polaris@beta
+yarn add @shopify/polaris@beta.0
```
Comment (laurkim) [inline polaris.shopify.com/content/new-design-language/index.md]:
```suggestion
yarn add @shopify/polaris@12.0.0-beta.0
```

**Review by lgriffee — APPROVED** [review]

**Pull request merged**